core/dbt/contracts/graph/unparsed.py

```diff
@@ -142,6 +138,62 @@ class HasConfig:
     config: Dict[str, Any] = field(default_factory=dict)
 
 
+NodeVersion = Union[str, float]
+
+
+@dataclass
+class UnparsedVersion(dbtClassMixin):
+    v: NodeVersion
+    defined_in: Optional[str] = None
+    description: str = ""
+    access: Optional[str] = None
+    config: Dict[str, Any] = field(default_factory=dict)
+    meta: Dict[str, Any] = field(default_factory=dict)
+    constraints: List[Dict[str, Any]] = field(default_factory=list)
+    docs: Docs = field(default_factory=Docs)
+    tests: Optional[List[TestDef]] = None
+    columns: Sequence[Union[dbt.helper_types.IncludeExclude, UnparsedColumn]] = field(
+        default_factory=list
+    )
```

Comment:
As discussed offline: I'd expect the default `columns` value for a version, if not otherwise set, to be `[include: "*"]`.

For example:
```yml
models:
  - name: my_model
    columns:
      - name: id
        data_type: int
    versions:
      - v: 2
      - v: 1
        columns: 
          - include: 'all'
          - name: id
            data_type: text
```
My expectation is that the `columns` in the top-level model definition == the `columns` in `v2`. In the current implementation, `v2` has `columns: []`.

(Not sure if that makes more sense to implement as the default here or in `get_columns_for_version` below)